airbyte/shared/sql_processor.py

```diff
@@ -124,6 +124,10 @@ def config_hash(self) -> str | None:
             )
         )
 
+    def get_create_table_extra_clauses(self) -> list[str]:
+        """Return a list of clauses to append on CREATE TABLE statements."""
+        return []
```

Comment:
@coderabbitai - This is not a free-form input but it's good to call out anyway. Do you have any suggestion to sanitize/pre-check the SQL?

Reply:
Having a startswith is not gonna be enough to forbid sql injections. If one of the database accepts multi-statement queries, one could do `DATA_RETENTION_TIME_IN_DAYS=1; INSERT INTO ... (SELECT ...); CREATE TABLE foo (...)` or similar things. That way even if there are multiple statements, they bypass it.

My take on this was to only let implementations of sql_processor specify their value.

However in a separator PR we could rework how query are being sent. Even though it's not a free form, we could still improve the security of the codebase against malicious actors.

```python
def _execute_sql(self, sql: str | TextClause | Executable) -> CursorResult
```

could be rewritten with the signature
```python
def _execute_sql(self, sql: str | TextClause | Executable, parameters: dict[str, Any]) -> CursorResult
```

and then we update the few queries being sent in the code to use parameters instead of string concatenation (`_add_column_to_table`, `_drop_temp_table`, `_append_temp_table_to_final_table` aren't safe if I'm not wrong, especially because there is no normalizer or anything in `_fully_qualified`).

What do you think?